### PR TITLE
fix(getLocalIdent): generate `localIdent` consistently across platforms

### PR DESCRIPTION
**Pull request by kerhong**
**What kind of change does this PR introduce?**
Made output of getLocalIdent platform independent

**Did you add tests for your changes?**
No

**Summary**
Made getLocalIdent normalize path separators to produce same output on all platforms.

**Does this PR introduce a breaking change?**
No

**Other information**
This solution was taken from https://github.com/webpack/loader-utils/blob/master/lib/interpolateName.js#L58

### CODE REVIEW

**Comment by kerhong** [comment]
I don't particularly like the changes I did to helpers.js in order to create this test. Perhaps you could suggest a better way to do the same test?

File: lib/getLocalIdent.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var path = require("path");
 module.exports = function getLocalIdent(loaderContext, localIdentName, localName, options) {
 	if(!options.context)
 		options.context = loaderContext.options && typeof loaderContext.options.context === "string" ? loaderContext.options.context : loaderContext.context;
-	var request = path.relative(options.context, loaderContext.resourcePath);
+	var request = path.relative(options.context, loaderContext.resourcePath).replace(/\\/g, "/").replace(/\.\.(\/)?/g, "_$1");
```
Comment (michael-ciniawsky) [inline lib/getLocalIdent.js]:
```
path.posix.relative
```

?  But not sure...

**Comment by michael-ciniawsky** [comment]
@kerhong This is likely to be breaking setups and we intent to remove CSS Modules from `css-loader` in the next major bump, so the CSSM implementation is in maintenance mode with 'critical' bugfixes only.

**Comment by jackfranklin** [comment]
@michael-ciniawsky is there a rough plan with how CSS Modules will be supported in webpack longer term?

I ask as we use them with css-loader heavily, and would like to start preparing for moving off that and onto something else :)

**Comment by michael-ciniawsky** [comment]
Sadly no current ETA and I always got distract by other things aswell :). I will try to streamline this effort and will post a RFC link here as soon as one is available

**Comment by michael-ciniawsky** [comment]
I'm declining this PR. As mentioned a while ago we can not be sure if this breaks existing setups or not and we want to remove CSS Modules as quickly as possible anyways since the current implementation is problematic in general

**Pull request closed without merging**